# Explore the Data for Parking Spots for Disabled Persons

## import the libraries

In [2]:
import geopandas as gpd
import geojson
import json
from OSMPythonTools.nominatim import Nominatim
from OSMPythonTools.overpass import Overpass, overpassQueryBuilder
from OSMPythonTools.api import Api

import seaborn as sns
import missingno as msno
%matplotlib inline

In [3]:
import pandas as pd
pd.options.display.max_columns = None
pd.options.display.max_rows = None
pd.options.display.max_colwidth = 200


In [4]:
from IPython.display import display, HTML

# utility function to display tables, views, etc side-by-side
css = """
.output {
    flex-direction: row;
}
"""

HTML('<style>{}</style>'.format(css))


## Load Data

In [5]:
df_accessible = gpd.read_file("../data/datathon/parking/parking_spaces_for_disabled_persons.geojson").astype({"ANZAHL": "int64"})
df_inhabitants = pd.read_csv("../data/datathon/demographics/demographics_age_of_inhabitants.csv", thousands=" ", decimal=",")
df_lor = gpd.read_file("../data/datathon/lor_friedrichshain_kreuzberg.geojson")

In [6]:
df_accessible.head()

ANZAHL       BEMERKUNG        ORTSTEIL    PLZ                    STANDORT  \
0       1     Durchgehend  Friedrichshain  10245          Alt - Stralau 32 b   
1       1  Mo-Fr 7-20.30h  Friedrichshain  10245            Alt - Stralau 61   
2       1     Mo-Fr 7-17h       Kreuzberg  10969        Alte Jakobstraße 149   
3       1    Mo-Fr 10-21h       Kreuzberg  10969  Axel-Springer-Straße 40/41   
4       1     Mo-Fr 8-18h       Kreuzberg  10969     Axel-Springer-Straße 44   

                    geometry  
0  POINT (13.47776 52.49258)  
1  POINT (13.46786 52.49573)  
2  POINT (13.39625 52.49921)  
3  POINT (13.39887 52.50772)  
4  POINT (13.39943 52.50848)

In [72]:
df_lor.head()

PLR_ID         PLR_NAME  BZR_ID                  BZR_NAME PGR_ID  \
0  02500831      Wismarplatz  025008  Frankfurter Allee Süd FK   0250   
1  02300315  Lausitzer Platz  023003     Nördliche Luisenstadt   0230   
2  02500832       Traveplatz  025008  Frankfurter Allee Süd FK   0250   
3  02100104      Moritzplatz  021001  Südliche  Friedrichstadt   0210   
4  02500726  Hausburgviertel  025007    Frankfurter Allee Nord   0250   

             PGR_NAME BEZ        FINHALT       STAND  \
0  Friedrichshain Ost  02  248991.830677  01.01.2021   
1       Kreuzberg Ost  02  350912.758939  01.01.2021   
2  Friedrichshain Ost  02  369572.692540  01.01.2021   
3      Kreuzberg Nord  02  380552.816915  01.01.2021   
4  Friedrichshain Ost  02  309450.362961  01.01.2021   

                                                                                                                                                                                                  geometry  
0  POLYGON ((396010.552 5819339.570, 395991.806 5819343.295, 395974.424 5819346.595, 395963.393 5819348.688, 395949.598 5819351.306, 395939.015 5819353.315, 395925.342 5819355.910, 395903.771 5819360...  
1  POLYGON ((393565.472 5817715.260, 393622.143 5817727.973, 393649.717 5817733.802, 393656.732 5817735.334, 393673.339 5817738.964, 393679.915 5817740.401, 393686.593 5817741.860, 393697.134 5817744...  
2  POLYGON ((396230.956 5819298.042, 396219.580 5819255.426, 396213.184 5819231.209, 396206.725 5819206.689, 396200.252 5819182.074, 396195.332 5819163.520, 396189.193 5819139.986, 396180.286 5819118...  
3  POLYGON ((391710.821 5818713.293, 391709.137 5818711.455, 391693.463 5818694.375, 391682.070 5818683.132, 391656.348 5818657.684, 391633.292 5818635.724, 391611.278 5818615.985, 391584.324 5818591...  
4  POLYGON ((394655.527 5820722.533, 394669.046 5820691.408, 394673.664 5820679.888, 394694.711 5820627.390, 394696.040 5820624.073, 394701.121 5820611.161, 394702.501 5820607.654, 394712.189 5820582...

In [7]:
# set as background map
def get_lor_map():
    bezirk_map = df_lor.explore(
        column="PLR_NAME",
        legend=True,
        legend_kwds=dict(colorbar=False),
        style_kwds=dict(opacity=0.4, fillOpacity=0.7),
        name="Friedrichshain-Kreuzberg"
    )
    return bezirk_map

In [161]:
get_lor_map()

In [9]:
total_accessible_parking_spots = df_accessible.ANZAHL.sum()
total_accessible_parking_spots

157

There are 157 parkink spots for disabled people in Friedrichshain Kreuzberg

now let's look at how many people are in Friedrichshain Kreuzberg

In [26]:
df_inhabitants.head()

lor_number   total  below age 6   6 - 15  15 - 18 18 - 27 27 - 45 45 - 55  \
0           1   6 257          368      508      172     834   2 309     793   
1           2   8 633          551      864      313   1 169   2 659   1 077   
2           3   5 385          374      531      155     635   1 732     693   
3           4   5 587          373      538      190     750   1 563     807   
4           5   9 527          563      940      294   1 144   2 899   1 263   

  55 - 65 65 an more of those female of those immigrants                lor  
0     629        644           2 912               2 507  Askanischer Platz  
1     907      1 093           4 270               3 004       Mehringplatz  
2     631        634           2 694               1 717   Am Berlin Museum  
3     719        647           2 744               2 074       Moritzplatz   
4   1 092      1 332           4 752               3 290      Prinzenstraße

In [10]:
total_inhabitants = df_inhabitants.total.sum()
total_inhabitants

288906

In [11]:
total_accessible_parking_spots / total_inhabitants * 100

0.05434293507230725

## Number of people with disabilities in Germany according to Statistisches Bundesamt

According to this article [link](https://www.destatis.de/DE/Presse/Pressemitteilungen/2020/06/PD20_230_227.html) there are around 9.8% of the German population living with a disability. This number is for the whole population of Germany and not for Friedrichshain Kreuzberg specifically

But 1 accessible parking spot for every 1840 inhabitants sounds way too few

In [39]:
total_inhabitants / total_accessible_parking_spots


1840.1656050955414

# Show accessible parking spots on the map

In [12]:
df_accessible.explore(
    m=get_lor_map()
)

## get number of accessible parking spots per lor

In [13]:
df_accessible.geometry = df_accessible.geometry.to_crs(25833)


In [14]:
df_accessible.geometry.within(df_lor.geometry)

/Users/lelischiestl/opt/anaconda3/envs/dssg/lib/python3.9/site-packages/geopandas/base.py:31: UserWarning: The indices of the two GeoSeries are different.
  warn("The indices of the two GeoSeries are different.")


0      False
1      False
2      False
3      False
4      False
5      False
6      False
7      False
8      False
9      False
10     False
11     False
12     False
13     False
14     False
15     False
16     False
17     False
18     False
19     False
20     False
21     False
22     False
23     False
24     False
25     False
26     False
27     False
28     False
29     False
30     False
31     False
32     False
33     False
34     False
35     False
36     False
37     False
38     False
39     False
40     False
41     False
42     False
43     False
44     False
45     False
46     False
47     False
48     False
49     False
50     False
51     False
52     False
53     False
54     False
55     False
56     False
57     False
58     False
59     False
60     False
61     False
62     False
63     False
64     False
65     False
66     False
67     False
68     False
69     False
70     False
71     False
72     False
73     False
74     False
75     False
76     False

In [46]:
df_test = df_accessible.sjoin(df_lor)

In [16]:
df_test.head()

ANZAHL       BEMERKUNG        ORTSTEIL    PLZ                    STANDORT  \
0        1     Durchgehend  Friedrichshain  10245          Alt - Stralau 32 b   
1        1  Mo-Fr 7-20.30h  Friedrichshain  10245            Alt - Stralau 61   
2        1     Mo-Fr 7-17h       Kreuzberg  10969        Alte Jakobstraße 149   
68       2  Mo-Fr 6-19.30h       Kreuzberg  10969           Oranienstraße 106   
3        1    Mo-Fr 10-21h       Kreuzberg  10969  Axel-Springer-Straße 40/41   

                          geometry  index_right    PLR_ID  \
0   POINT (396651.507 5816915.423)           16  02500836   
1   POINT (395986.790 5817280.079)           16  02500836   
2   POINT (391134.205 5817773.345)           24  02100103   
68  POINT (391438.749 5818569.490)           24  02100103   
3   POINT (391332.977 5818715.085)           33  02100102   

               PLR_NAME  BZR_ID                  BZR_NAME PGR_ID  \
0   Stralauer Halbinsel  025008  Frankfurter Allee Süd FK   0250   
1   Stralauer Halbinsel  025008  Frankfurter Allee Süd FK   0250   
2      Am Berlin Museum  021001  Südliche  Friedrichstadt   0210   
68     Am Berlin Museum  021001  Südliche  Friedrichstadt   0210   
3          Mehringplatz  021001  Südliche  Friedrichstadt   0210   

              PGR_NAME BEZ       FINHALT       STAND  
0   Friedrichshain Ost  02  1.117020e+06  01.01.2021  
1   Friedrichshain Ost  02  1.117020e+06  01.01.2021  
2       Kreuzberg Nord  02  4.132675e+05  01.01.2021  
68      Kreuzberg Nord  02  4.132675e+05  01.01.2021  
3       Kreuzberg Nord  02  6.244703e+05  01.01.2021

In [54]:
df_acc_in_lor = df_test[["index_right", "ANZAHL"]].rename(columns ={"index_right": "lor", "ANZAHL": "count"})
df_acc_per_lor= df_acc_in_lor.groupby("lor").sum()

In [55]:
df_acc_in_lor.sort_values("lor").head()

lor  count
17    0      1
29    0      2
21    0      2
96    1      1
95    1      1

In [94]:
df_acc_per_lor

count
lor       
0        5
1        2
3        3
5        4
6        3
7        9
9        6
10       8
12       7
13       5
14       2
15       6
16       2
17       6
18       5
19       8
20       1
21       1
22       5
23       6
24       3
25       8
26       3
27       4
29       4
30       2
31       3
32       5
33      16
34       5
35       9

# looking at the time availability of accessible parking spots

In [58]:
df_accessible

ANZAHL                                BEMERKUNG        ORTSTEIL    PLZ  \
0         1                              Durchgehend  Friedrichshain  10245   
1         1                           Mo-Fr 7-20.30h  Friedrichshain  10245   
2         1                              Mo-Fr 7-17h       Kreuzberg  10969   
3         1                             Mo-Fr 10-21h       Kreuzberg  10969   
4         1                              Mo-Fr 8-18h       Kreuzberg  10969   
5         1                              Mo-Fr 8-17h  Friedrichshain  10247   
6         1                              Mo-Fr 8-16h  Friedrichshain  10249   
7         1                              Durchgehend  Friedrichshain  10249   
8         1                              Durchgehend       Kreuzberg  10961   
9         1                              Mo-Fr 8-18h       Kreuzberg  10963   
10        1                              Durchgehend       Kreuzberg  10961   
11        3                              Durchgehend       Kreuzberg  10961   
12        1                              Mo-Fr 8-19h       Kreuzberg  10961   
13        2                              Mo-Fr 8-18h       Kreuzberg  10961   
14        1                              Durchgehend       Kreuzberg  10967   
15        1                              Durchgehend       Kreuzberg  10967   
16        2                       Sa 8-16h, So 9-18h  Friedrichshain  10245   
17        1                          (täglich) 9-22h  Friedrichshain  10245   
18        1                              Mo-Fr 7-17h       Kreuzberg  10969   
19        1                              Durchgehend  Friedrichshain  10243   
20        1                             Mo-Fr 12-18h       Kreuzberg  10997   
21        2                              Mo-Fr 8-18h  Friedrichshain  10247   
22        1                             Mo-Fr  7-16h       Kreuzberg  10999   
23        2                  Mo - Fr 9-18h, Sa 9-14h  Friedrichshain  10247   
24        1                    Mo-Do 7-19h, Fr 7-16h  Friedrichshain  10247   
25        1                             Do-So 15-22h  Friedrichshain  10247   
26        2                              Mo-Fr 9-16h       Kreuzberg  10969   
27        1                              Mo-Fr 7-16h  Friedrichshain  10249   
28        1                    Mo-Fr 9-18h, Sa 9-14h  Friedrichshain  10247   
29        2                              Mo-Fr 8-20h  Friedrichshain  10247   
30        2                             Mo-Fr  6-18h       Kreuzberg  10999   
31        1                              Mo-Fr 9-21h       Kreuzberg  10961   
32        1                             Mo-Fr  8-18h       Kreuzberg  10961   
33        1                              Durchgehend       Kreuzberg  10997   
34        2                              Mo-Fr 9-18h       Kreuzberg  10963   
35        1                           Mo-Fr 8.30-18h       Kreuzberg  10965   
36        2                         (täglich) 19-23h       Kreuzberg  10963   
37        1                              Mo-Fr 8-18h  Friedrichshain  10243   
38        2                              Mo-Fr 7-20h  Friedrichshain  10245   
39        1                           werktags 7-18h  Friedrichshain  10243   
40        1                             Mo-Fr 13-19h  Friedrichshain  10243   
41        1                              Mo-Fr 8-19h       Kreuzberg  10965   
42        1                              Durchgehend       Kreuzberg  10965   
43        1                              Durchgehend       Kreuzberg  10965   
44        1                    Mo-Do 8-18h, Fr 8-13h       Kreuzberg  10967   
45        2                              Mo-Fr 8-20h       Kreuzberg  10997   
46        1                              Mo-Fr 7-18h       Kreuzberg  10965   
47        1                    Mo-Fr 8-18h, Sa 9-16h  Friedrichshain  10243   
48        1                              Mo-Fr 9-18h  Friedrichshain  10243   
49        1                              Mo Fr 8-16h   

In [59]:
df_accessible.BEMERKUNG.unique()

array(['Durchgehend', 'Mo-Fr 7-20.30h', 'Mo-Fr 7-17h', 'Mo-Fr 10-21h',
       'Mo-Fr 8-18h', 'Mo-Fr 8-17h', 'Mo-Fr 8-16h', 'Mo-Fr 8-19h',
       'Sa 8-16h, So 9-18h', '(täglich) 9-22h', 'Mo-Fr 12-18h',
       'Mo-Fr  7-16h', 'Mo - Fr 9-18h, Sa 9-14h', 'Mo-Do 7-19h, Fr 7-16h',
       'Do-So 15-22h', 'Mo-Fr 9-16h', 'Mo-Fr 7-16h',
       'Mo-Fr 9-18h, Sa 9-14h', 'Mo-Fr 8-20h', 'Mo-Fr  6-18h',
       'Mo-Fr 9-21h', 'Mo-Fr  8-18h', 'Mo-Fr 9-18h', 'Mo-Fr 8.30-18h',
       '(täglich) 19-23h', 'Mo-Fr 7-20h', 'werktags 7-18h',
       'Mo-Fr 13-19h', 'Mo-Do 8-18h, Fr 8-13h', 'Mo-Fr 7-18h',
       'Mo-Fr 8-18h, Sa 9-16h', 'Mo Fr 8-16h', 'Mo-Fr 6-18h',
       '(täglich) 9-20h', 'Mo-Do 8-17h, Fr 8-15h', 'Mo-Fr 6.30-20h',
       'Mo-Fr 9-18h, Sa 9-13h', 'Mo-Fr 7-19h', 'Mo-Fr 9-20h',
       'Mo-Fr 6-19.30h', 'Mo-Fr 7-15h', 'Mo-Fr 10-20h',
       'Mo-Fr 6-8h, 13-15h', 'Mo-So 9-21h', 'Mo-Fr 7.30-18h',
       'Mo-Fr 8-19h, Sa 8-16h', 'Sonn- u. Feiertags 10-13h',
       'Mo-Fr 8.30-18h, Sa 9-11h', 'Mo-Fr

In [61]:
df_accessible[["ANZAHL", "BEMERKUNG"]].groupby("BEMERKUNG").count()

ANZAHL
BEMERKUNG                                      
(täglich) 18-23h                              1
(täglich) 19-23h                              1
(täglich) 9-20h                               1
(täglich) 9-22h                               1
(täglich) 9-23h                               1
Do-So 15-22h                                  1
Durchgehend                                  24
Mi 18-21h, So 9-12h und 18-21h                1
Mo - Fr 9-18h, Sa 9-14h                       1
Mo Fr 8-16h                                   1
Mo,Di,Do,Fr 8-14h, Mi 14-18h                  1
Mo-Do 7-19h, Fr 7-16h                         1
Mo-Do 8-17h, Fr 8-15h                         1
Mo-Do 8-18h, Fr 8-13h                         1
Mo-Fr  6-18h                                  1
Mo-Fr  7-16h                                  1
Mo-Fr  8-18h                                  1
Mo-Fr 10-17h                                  1
Mo-Fr 10-20h                                  1
Mo-Fr 10-21h                                  1
Mo-Fr 12-18h                                  1
Mo-Fr 13-19h                                  1
Mo-Fr 6-18h                                   4
Mo-Fr 6-19.30h                                1
Mo-Fr 6-8h, 13-15h                            1
Mo-Fr 6.30-20h                                1
Mo-Fr 7-15h                                   1
Mo-Fr 7-16h                                   1
Mo-Fr 7-17h                                   3
Mo-Fr 7-18h                                   2
Mo-Fr 7-19h                                   2
Mo-Fr 7-20.30h                                1
Mo-Fr 7-20h                                   2
Mo-Fr 7.30-18h                                1
Mo-Fr 7.30-19.30h                             1
Mo-Fr 8-16h                                   3
Mo-Fr 8-17h                                   3
Mo-Fr 8-18h                                  11
Mo-Fr 8-18h + Zusatz Parkscheibe 2h           1
Mo-Fr 8-18h, Sa 9-16h                         1
Mo-Fr 8-19h                                   2
Mo-Fr 8-19h, Sa 8-16h                         1
Mo-Fr 8-20h                                   2
Mo-Fr 8-21h                                   1
Mo-Fr 8.30-18h                                1
Mo-Fr 8.30-18h, Sa 9-11h                      1
Mo-Fr 9-16h                                   2
Mo-Fr 9-18h                                   5
Mo-Fr 9-18h, Sa 9 -14h                        1
Mo-Fr 9-18h, Sa 9-13h                         1
Mo-Fr 9-18h, Sa 9-14h                         3
Mo-Fr 9-20h                                   2
Mo-Fr 9-21h                                   1
Mo-So 9-20h                                   1
Mo-So 9-21h                                   1
Sa 8-16h, So 9-18h                            1
Sonn- u. Feiertags 10-13h                     1
werktags 7-18h                                1
während der Öffnungszeiten Bezirksamtes       1

In [71]:
df_accessible[df_accessible.BEMERKUNG.str.match(r"^Mo-Fr")]

ANZAHL                            BEMERKUNG        ORTSTEIL    PLZ  \
1         1                       Mo-Fr 7-20.30h  Friedrichshain  10245   
2         1                          Mo-Fr 7-17h       Kreuzberg  10969   
3         1                         Mo-Fr 10-21h       Kreuzberg  10969   
4         1                          Mo-Fr 8-18h       Kreuzberg  10969   
5         1                          Mo-Fr 8-17h  Friedrichshain  10247   
6         1                          Mo-Fr 8-16h  Friedrichshain  10249   
9         1                          Mo-Fr 8-18h       Kreuzberg  10963   
12        1                          Mo-Fr 8-19h       Kreuzberg  10961   
13        2                          Mo-Fr 8-18h       Kreuzberg  10961   
18        1                          Mo-Fr 7-17h       Kreuzberg  10969   
20        1                         Mo-Fr 12-18h       Kreuzberg  10997   
21        2                          Mo-Fr 8-18h  Friedrichshain  10247   
22        1                         Mo-Fr  7-16h       Kreuzberg  10999   
26        2                          Mo-Fr 9-16h       Kreuzberg  10969   
27        1                          Mo-Fr 7-16h  Friedrichshain  10249   
28        1                Mo-Fr 9-18h, Sa 9-14h  Friedrichshain  10247   
29        2                          Mo-Fr 8-20h  Friedrichshain  10247   
30        2                         Mo-Fr  6-18h       Kreuzberg  10999   
31        1                          Mo-Fr 9-21h       Kreuzberg  10961   
32        1                         Mo-Fr  8-18h       Kreuzberg  10961   
34        2                          Mo-Fr 9-18h       Kreuzberg  10963   
35        1                       Mo-Fr 8.30-18h       Kreuzberg  10965   
37        1                          Mo-Fr 8-18h  Friedrichshain  10243   
38        2                          Mo-Fr 7-20h  Friedrichshain  10245   
40        1                         Mo-Fr 13-19h  Friedrichshain  10243   
41        1                          Mo-Fr 8-19h       Kreuzberg  10965   
45        2                          Mo-Fr 8-20h       Kreuzberg  10997   
46        1                          Mo-Fr 7-18h       Kreuzberg  10965   
47        1                Mo-Fr 8-18h, Sa 9-16h  Friedrichshain  10243   
48        1                          Mo-Fr 9-18h  Friedrichshain  10243   
50        1                          Mo-Fr 6-18h       Kreuzberg  10999   
51        1                          Mo-Fr 6-18h       Kreuzberg  10999   
52        1                          Mo-Fr 7-17h  Friedrichshain  10243   
54        2                          Mo-Fr 8-18h       Kreuzberg  10969   
58        1                          Mo-Fr 9-18h       Kreuzberg  10967   
59        1                       Mo-Fr 6.30-20h  Friedrichshain  10249   
61        1                          Mo-Fr 8-18h       Kreuzberg  10961   
62        1                Mo-Fr 9-18h, Sa 9-13h       Kreuzberg  10961   
63        1                          Mo-Fr 9-18h       Kreuzberg  10963   
65        1                          Mo-Fr 7-19h  Friedrichshain  10245   
66        1                          Mo-Fr 9-20h       Kreuzberg  10963   
67        1                          Mo-Fr 9-20h       Kreuzberg  10997   
68        2                       Mo-Fr 6-19.30h       Kreuzberg  10969   
69        1                          Mo-Fr 7-19h       Kreuzberg  10969   
71        1                          Mo-Fr 7-15h       Kreuzberg  10999   
72        1                          Mo-Fr 8-17h       Kreuzberg  10969   
73        1                         Mo-Fr 10-20h  Friedrichshain  10243   
74        4                   Mo-Fr 6-8h, 13-15h       Kreuzberg  10999   
75        2                          Mo-Fr 8-18h  Friedrichshain  10247   
76        4                          Mo-Fr 7-18h  Friedrichshain  10247   
77        1                Mo-Fr 9-18h, Sa 9-14h  Friedrichshain  10247   
78        1                Mo-Fr 9-18h, Sa 9-14h  Friedrichshain  10247   
79        1                   

# Gif the map

In [72]:
df_accessible.explore()

In [80]:
df_accessible[df_accessible.BEMERKUNG == "Durchgehend"].explore(m=get_lor_map())

In [79]:
fig = df_accessible.explore(m=get_lor_map())
fig

In [87]:
df_accessible.explore(
    legend=True,
    name="Durchgehend",
    legend_kwds={"caption":"test"}
)

In [92]:
def test_map():
    tmap = df_accessible.explore(
        column="ANZAHL",
        legend=True,
        legend_kwds=dict(caption="TEST)-}{(}uaenten", colorbar=False)
    )
    return tmap

In [93]:
test_map()

In [95]:
import folium

In [96]:
m2=folium.Map(location=loc)
folium.TileLayer('Stamen Terrain').add_to(m2)
folium.TileLayer('Stamen Toner').add_to(m2)
folium.TileLayer('Stamen Water Color').add_to(m2)
folium.TileLayer('cartodbpositron').add_to(m2)
folium.TileLayer('cartodbdark_matter').add_to(m2)
folium.LayerControl().add_to(m2)

m2

NameError: name 'loc' is not defined

In [151]:


import folium

mapa = folium.Map([52.51, 13.45],
                  zoom_start=13.3,
                  tiles='cartodbpositron')

points = folium.features.GeoJson(df_accessible.geometry)

mapa.add_children(points)

folium.TileLayer()
mapa



/var/folders/0n/_k8_jq_j53g_52n46cdsgp540000gn/T/ipykernel_96511/1827669630.py:9: FutureWarning: Method `add_children` is deprecated. Please use `add_child` instead.
  mapa.add_children(points)


import folium
m2=folium.Map(location=loc)
folium.TileLayer('Stamen Terrain').add_to(m2)
folium.TileLayer('Stamen Toner').add_to(m2)
folium.TileLayer('Stamen Water Color').add_to(m2)
folium.TileLayer('cartodbpositron').add_to(m2)
folium.TileLayer('cartodbdark_matter').add_to(m2)
folium.LayerControl().add_to(m2)

m2


In [ ]:
m2 = folium.Map([52.51, 13.45],
                  zoom_start=13.3,
                  tiles='cartodbpositron')
folium.TileLayer('Stamen Terrain').add_to(m2)
folium.TileLayer('Stamen Toner').add_to(m2)
folium.TileLayer('Stamen Water Color').add_to(m2)
folium.TileLayer('cartodbpositron').add_to(m2)
folium.TileLayer('cartodbdark_matter').add_to(m2)
folium.LayerControl().add_to(m2)

m2

In [ ]:
points = folium.features.GeoJson(df_accessible.geometry)

m2.add_child(points)

In [162]:
m4 = df_accessible.explore()

In [180]:
m7 = folium.Map(
    location=[52.51, 13.45],
    zoom_start=13.3,
    tiles="openstreetmap"
)

points = folium.features.GeoJson(
    df_accessible.geometry,
    marker=folium.Circle()
)
points.add_to(m7)

folium.features.GeoJson(
    df_test.geometry,
    marker=folium.CircleMarker()
).add_to(m7)
folium.LayerControl().add_to(m7)

# folium.features.GeoJson(df_accessible.geometry)

m7


In [171]:
m7

In [181]:
df_with_timestamp = df_accessible

In [183]:
import datetime

In [187]:
df_with_timestamp["time"] = pd.to_datetime("2021-01-01")

In [192]:
df2_with_timestamp = df_accessible[df_accessible.BEMERKUNG == "Durchgehend"]
df2_with_timestamp["time"] = pd.to_datetime("2021-02-01")
df2_with_timestamp


/Users/lelischiestl/opt/anaconda3/envs/dssg/lib/python3.9/site-packages/geopandas/geodataframe.py:1351: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  super().__setitem__(key, value)


ANZAHL    BEMERKUNG        ORTSTEIL    PLZ  \
0         1  Durchgehend  Friedrichshain  10245   
7         1  Durchgehend  Friedrichshain  10249   
8         1  Durchgehend       Kreuzberg  10961   
10        1  Durchgehend       Kreuzberg  10961   
11        3  Durchgehend       Kreuzberg  10961   
14        1  Durchgehend       Kreuzberg  10967   
15        1  Durchgehend       Kreuzberg  10967   
19        1  Durchgehend  Friedrichshain  10243   
33        1  Durchgehend       Kreuzberg  10997   
42        1  Durchgehend       Kreuzberg  10965   
43        1  Durchgehend       Kreuzberg  10965   
55        3  Durchgehend  Friedrichshain  10243   
56        1  Durchgehend       Kreuzberg  10961   
60        3  Durchgehend  Friedrichshain  10249   
64        1  Durchgehend       Kreuzberg  10965   
70        1  Durchgehend       Kreuzberg  10999   
82        1  Durchgehend       Kreuzberg  10999   
89        1  Durchgehend  Friedrichshain  10245   
90        1  Durchgehend       Kreuzberg  10963   
96        1  Durchgehend       Kreuzberg  10999   
100       2  Durchgehend  Friedrichshain  10245   
102       3  Durchgehend       Kreuzberg  10999   
108       1  Durchgehend       Kreuzberg  10997   
113       2  Durchgehend  Friedrichshain  10243   

                                       STANDORT  \
0                            Alt - Stralau 32 b   
7                               Barnimstraße 18   
8                             Bergmannstraße 25   
10                               Blücherplatz 1   
11                               Blücherplatz 3   
14                               Böckhstraße 16   
15   Böckhstraße Höhe 43 / Ecke Graefestraße 83   
19          Erich-Steinfurt-Straße (Ostbahnhof)   
33                               Görlitzer Park   
42                        Hagelberger Straße 34   
43                         Hagelberger Straße 8   
55                      Marchlewskistraße 4 / 6   
56                     Marheinekeplatz ggü. 3-4   
60        Max-Flettling-Platz (Ernst-Zinna-Weg)   
64                             Möckernstraße 69   
70            Oranienstraße 192 (Heinrichplatz)   
82                     Reichenberger Straße 130   
89     Rudolfstraße (U-Bahnhof Warschauer Str.)   
90                         Ruhlsdorfer Straße 2   
96                          Skalitzer Straße 36   
100                  Warschauer Platz U-Bahnhof   
102         Wiener Straße 59 h (Spreewaldplatz)   
108                              Görlitzer Park   
113          Frankfurter Allee 14a (Bibliothek)   

                           geometry       time  
0    POINT (396651.507 5816915.423) 2021-02-01  
7    POINT (392988.035 5820722.338) 2021-02-01  
8    POINT (391009.718 5816646.440) 2021-02-01  
10   POINT (390868.257 5817540.812) 2021-02-01  
11   POINT (390801.135 5817540.902) 2021-02-01  
14   POINT (392678.740 5817112.293) 2021-02-01  
15   POINT (392642.847 5817139.390) 2021-02-01  
19   POINT (393738.623 5819076.499) 2021-02-01  
33   POINT (393843.873 5817566.670) 2021-02-01  
42   POINT (390067.298 5816952.026) 2021-02-01  
43   POINT (390348.044 5816929.325) 2021-02-01  
55   POINT (394310.035 5819607.890) 2021-02-01  
56   POINT (391152.435 5816662.782) 2021-02-01  
60   POINT (393966.570 5820632.029) 2021-02-01  
64   POINT (389958.884 5816814.581) 2021-02-01  
70   POINT (392927.667 5817842.346) 2021-02-01  
82   POINT (393405.872 5817336.979) 2021-02-01  
89   POINT (394795.131 5818361.094) 2021-02-01  
90   POINT (390559.145 5817580.427) 2021-02-01  
96   POINT (393165.685 5817679.023) 2021-02-01  
100  POINT (394761.895 5818336.117) 2021-02-01  
102  POINT (393584.588 5817505.302) 2021-02-01  
108  POINT (393817.301 5817581.420) 2021-02-01  
113  POINT (395320.306 5819424.647) 2021-02-01

In [195]:
df_with_timestamp = pd.concat([df_with_timestamp, df2_with_timestamp])

In [202]:
from folium.plugins import TimestampedGeoJson

In [211]:
m8 = folium.Map(
    location=[52.51, 13.45],
    zoom_start=13.3,
    tiles="openstreetmap"
)

start_geojson = [
    {
        "type": "Feature",
        "geometry": {
            "type": "Point",
            "coordinates": [52.5, 13.5]
        },
        "properties": {"time": "2021-01-01", "style": {"color": ""}, "icon": "circle", "iconstyle": {"fillColor": "#586065", "fillOpacity": 0.8, "stroke": "true", "radius": 5}}},
    {"type": "Feature", "geometry": {"type": "Point", "coordinates": [52.6, 13.5]}, "properties": {"time": "2021-02-01"}},
]
TimestampedGeoJson(start_geojson,period = 'PT1H',
                  duration = 'P1M',
                  transition_time = 1000,
                  auto_play = True).add_to(m8)
m8

In [224]:
df_accessible.head()

ANZAHL       BEMERKUNG        ORTSTEIL    PLZ                    STANDORT  \
0       1     Durchgehend  Friedrichshain  10245          Alt - Stralau 32 b   
1       1  Mo-Fr 7-20.30h  Friedrichshain  10245            Alt - Stralau 61   
2       1     Mo-Fr 7-17h       Kreuzberg  10969        Alte Jakobstraße 149   
3       1    Mo-Fr 10-21h       Kreuzberg  10969  Axel-Springer-Straße 40/41   
4       1     Mo-Fr 8-18h       Kreuzberg  10969     Axel-Springer-Straße 44   

                         geometry       time  
0  POINT (396651.507 5816915.423) 2021-01-01  
1  POINT (395986.790 5817280.079) 2021-01-01  
2  POINT (391134.205 5817773.345) 2021-01-01  
3  POINT (391332.977 5818715.085) 2021-01-01  
4  POINT (391373.243 5818799.294) 2021-01-01

In [230]:
berlin_map = folium.Map(
    location=[52.51, 13.45],
    zoom_start=13.3,
    tiles="openstreetmap"
)

parking_spot_1 = {
    "type": "Feature",
    "geometry": {
        "type": "Point",
        "coordinates": [
            52, 13
        ] 
    },
    "properties": {
        "style": {
            "color": "red"
        },
        "times": ['2015-07-22T00:00:00', '2015-08-22T00:00:00',
                  '2015-09-22T00:00:00', '2015-10-22T00:00:00',
                  '2015-11-22T00:00:00', '2015-12-22T00:00:00',
                  '2016-11-22T00:00:00', '2020-11-22T00:00:00']
    }
}

TimestampedGeoJson(
    {'type': 'FeatureCollection', 'features': [parking_spot_1]},
    period='P1M',
    duration='P1M',
    auto_play=False,
    loop=False,
    loop_button=True,
    date_options='YYYY/MM/DD',
).add_to(berlin_map)

berlin_map

In [ ]:
import folium
from folium.plugins import TimestampedGeoJson

m = folium.Map(location=[52.467697, -2.548828], zoom_start=6)

polygon_1 = {
    'type': 'Feature',
    'geometry': {
        'type': 'MultiPolygon',
        'coordinates': [((
             (-2.548828, 51.467697),
             (-0.087891, 51.536086),
             (-1.516113, 53.800651),
             (-6.240234, 53.383328),
        ),)] * 6, # duplication for matching 6 timestamps
    },
    'properties': {
        'style': {
            'color': 'blue',
        },
        'times': ['2015-07-22T00:00:00', '2015-08-22T00:00:00',
                  '2015-09-22T00:00:00', '2015-10-22T00:00:00',
                  '2015-11-22T00:00:00', '2015-12-22T00:00:00']
    }
}

polygon_2 = {
    'type': 'Feature',
    'geometry': {
        'type': 'MultiPolygon',
        'coordinates': [((
             (-3.548828, 50.467697),
             (-1.087891, 50.536086),
             (-2.516113, 52.800651),
             (-7.240234, 52.383328),
        ),)] * 2, # duplication for matching 2 timestamps
    },
    'properties': {
        'style': {
            'color': 'yellow',
        },
        'times': ['2015-07-22T00:00:00', '2015-08-22T00:00:00']
    }
}

TimestampedGeoJson(
    {'type': 'FeatureCollection', 'features': [polygon_1, polygon_2]},
    period='P1M',
    duration='P1M',
    auto_play=False,
    loop=False,
    loop_button=True,
    date_options='YYYY/MM/DD',
).add_to(m)
    
m
